pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -417,20 +423,24 @@ def get_checksum_types(self, **kwargs):
 
         return repomd_checksum_types, primary_checksum_types
 
-    def test_unspecified_checksum_types(self):
-        """Sync and publish an RPM repository and verify the checksum."""
-        repomd_checksum_types, primary_checksum_types = self.get_checksum_types()
+    def test_on_demand_unspecified_checksum_types(self):
+        """Sync and publish an RPM repository and verify the checksum types."""
+        repomd_checksum_types, primary_checksum_types = self.get_checksum_types(
+            fixture_url=RPM_SHA512_FIXTURE_URL, policy="on_demand"
+        )
 
         for repomd_type, repomd_checksum_type in repomd_checksum_types.items():
             self.assertEqual(repomd_checksum_type, "sha256")
```

Comment:
I would expect metadata to be sha512 as well.

Reply:
Yup, I know, I shouldn't have marked this ready quite yet.  Sorry!

Reply:
I'll make any further changes in a new commit so it's easy to review

Reply:
I ended up needing more changes than I though.  It's all merged into one.